sql/src/test/java/org/apache/druid/sql/calcite/CalciteQueryTest.java

```diff
@@ -15336,4 +15336,34 @@ public void testFilterParseLongNullable()
         ImmutableList.of(new Object[]{NullHandling.sqlCompatible() ? 4L : 0L})
     );
   }
+
+  @Test
+  public void testLatestByAggregatorOnSecondaryTimestampGroupBy()
```

Comment:
please once ensure that the test is calling the `combiningFactory` of the aggregator, since that is the codepath that contains the bug. I'm not sure whether this does that since this datasource contains a single segment iirc.

Reply:
The codepath that has the bug is the `aggregate` of the factory which would also be called by the `combiningFactory` of the aggregator. In the test case that is added here, without pushing down the null check, the results would have been all nulls.

Reply:
while it doesn't go through combining factory at query time, it still ends up testing the right scenario because this query goes through group by merge query runner which runs on the result rows (row adapter) where the value selector there gets the pairs and time selector is nil because the referenced column isn't present at that point, so its probably ok. A test in the native layer probably would have been a bit clearer, but at least this provides some coverage

Reply:
ok yes, I missed that merge query runner would be exercised even for a single segment. :+1: